src/main/java/sirius/kernel/cache/Cache.java

```diff
@@ -152,11 +152,35 @@
      * Being all string based, this also works for coherent caches by simply broadcasting two short string values.
      *
      * @param discriminator the name of the remover
-     * @param test         the predicate which determines if a given entry matches a given test input
+     * @param test          the predicate which determines if a given entry matches a given test input
      * @return the cache itself for fluent method calls
      */
     Cache<K, V> addRemover(@Nonnull String discriminator, @Nonnull BiPredicate<String, CacheEntry<K, V>> test);
 
+    /**
+     * Starts building a remove handler identified with the given <tt>discriminator</tt>.
+     * <p>
+     * Contrary to {@link #addRemover(String, BiPredicate)}, this method returns a {@link CacheRemoverBuilder}
+     * that allows to build the remover with a fluent API.
+     *
+     * @param discriminator the name of the remover
+     * @return a {@link CacheRemoverBuilder} to build the remover
+     * @see #addRemover(String, BiPredicate) addRemover
+     */
+    CacheRemoverBuilder<K, V, CacheEntry<K, V>> addRemover(@Nonnull String discriminator);
+
+    /**
+     * Convenience function for {@code addRemover(discriminator).map(CacheEntry::getValue)}.
+     *
+     * @param discriminator the name of the remover
+     * @return a {@link CacheRemoverBuilder} to build the remover
+     * @see #addRemover(String)
+     * @see #addRemover(String, BiPredicate)
+     */
+    default CacheRemoverBuilder<K, V, V> addValueBasedRemover(@Nonnull String discriminator) {
```

Comment:
Although certainly not wrong - IMHO default should mainly be used for legacy "fixes" and we should move this into the impl - even if this means that it'll be a bit redundant...